<a href="https://colab.research.google.com/github/hmarathe420/Diabetes_Prediction/blob/main/Diabetes_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Project Name**    - Diabetes Prediction



##### **Project Type**    - Classification
##### **Contribution**    - Individual
##### **Name**            - Harshal Marathe

## **Project Summary -**

As an AIML intern at InternPE, I undertook a project focused on Diabetes Classification, leveraging machine learning techniques to predict the likelihood of individuals developing diabetes based on their medical and lifestyle attributes.

**Project Components:**

* Data Understanding: The project began with a comprehensive exploration of the dataset containing medical and lifestyle attributes. This step involved understanding the data's structure, variables, and underlying patterns.

* Exploratory Data Analysis (EDA) and Visualization: Through exploratory data analysis, I delved deeper into the dataset's characteristics, uncovering insights, correlations, and potential trends. Visualizations were used to present key findings effectively.

* Classification Algorithms: To predict diabetes, I employed a variety of classification algorithms, including Logistic Regression, Random Forest, and others. Each algorithm was implemented and fine-tuned to suit the specific problem.

* Model Performance Metrics: For evaluating model performance, I utilized a range of metrics such as accuracy, precision, recall, F1-score, and ROC-AUC. These metrics provided a comprehensive assessment of how well the models classified diabetes cases.

* Best Performer Model Selection: After evaluating different models, I identified the best-performing algorithm based on its overall performance metrics. This selection was crucial in determining the final model for deployment.

* Conclusion: The project's culmination led to a conclusive summary of findings and outcomes. I highlighted the strengths and limitations of the classification models and provided insights into the implications of the predictive results.

**Project Deliverables:**

Comprehensive data understanding and EDA documentation highlighting the dataset's characteristics and trends.
Visualizations to showcase key insights gained during EDA.
Implementation and evaluation of various classification algorithms, documented in code and accompanied by performance metrics.
Selection of the best-performing classification model, justified with supporting metrics and reasoning.
A conclusive project conclusion summarizing findings, model performance, and potential real-world implications.

**Project Impact:**

This project's significance lies in its potential to contribute to early diabetes detection, allowing individuals to proactively manage their health. The insights gained from EDA and the performance of various classification algorithms provide a valuable foundation for informed decision-making in healthcare settings.



## **GitHub Link -**

Provide your GitHub Link here.

## **Problem Statement**


Diabetes is a prevalent chronic health condition with significant implications for individuals' well-being. Early detection of diabetes can lead to timely interventions and improved management. Leveraging machine learning techniques to classify individuals' diabetes risk based on medical and lifestyle attributes has emerged as a promising approach.

As an AIML intern at InternPE, my task with developing a robust machine learning model that classifies individuals as either diabetic or non-diabetic based on their medical and lifestyle attributes. The goal is to create an accurate classification system that assists in identifying individuals who may be at risk of developing diabetes.

## **Data Collection & Understanding**

In [1]:
# Importing Required Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
# Mounting Google Drive with Google colab
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# loading dataset from drive
df = pd.read_csv("/content/drive/MyDrive/Diabetes Prediction/diabetes.csv")

In [5]:
# Top 5 rows of diabetes dataset
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [6]:
# dataset information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [7]:
# Describing the dataset
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [8]:
# shape of dataset
df.shape

(768, 9)

In [10]:
# number of duplicated rows of dataset
df.duplicated().sum()

0

In [12]:
# null values in dataset
df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

From the above we can conclude that the diabetes dataset has 768 rows and 9 columns. and there is no any duplicated rows and null values present in the diabetes dataset.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your Machine Learning Capstone Project !!!***